File: easy/best_time_to_buy_and_sell_stock.rb
def max_profit(prices)
  max = 0
  buy, sell = 0, 1
  
  while sell < prices.length
    if prices[sell] > prices[buy]
      max = [prices[sell] - prices[buy], max].max
    else
      buy = sell
    end
    sell += 1
  end
  max
end

File: easy/best_time_to_buy_and_sell_stock.js
const maxProfit = prices => {
  let max = 0;
  let buy = 0;
  let sell = 1;

  while (sell < prices.length) {
    if (prices[sell] > prices[buy]) {
      max = Math.max(prices[sell] - prices[buy], max)
    } else {
      buy = sell
    }
    sell++;
  }
  return max;
}